### ISSUE TITLE
Method too large

### ISSUE THREAD

**Issue by mgroth0** [Kotlin/kotlinx-kover#458]
**Describe the bug**

On using kotlinx-kover in a project that has a dependency on the amazon s3 kotlin library, I get error output.

**Errors**

```
[2023.09.13 18:09:13] (Coverage): Error during class instrumentation: aws.sdk.kotlin.services.s3.endpoints.DefaultEndpointProvider: org.jetbrains.coverage.org.objectweb.asm.MethodTooLargeException: Method too large: aws/sdk/kotlin/services/s3/endpoints/DefaultEndpointProvider.resolveEndpoint (Laws/sdk/kotlin/services/s3/endpoints/EndpointParameters;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
```

**Expected behavior**

No error output. Otherwise, when I search for this error online I should be able to find some suggestions on what I should do. However, I cannot find anything.

**Reproducer**

Hopefully it is as easy as using kover on any module that depends on the kotlin-aws-s3 library.

**Environment**
 - Kover Gradle Plugin version: 0.7.3
 - Gradle version: 8.3
 - Kotlin project type: Kotlin/Jvm
 - Coverage Toolset (if customized in build script): Kover

**Comment by shanshin** [comment]
Hi,

> Also, just curious, why is the plugin trying to instrument a class from a library? Isn't coverage supposed to just collect data related to my own classes?

instrumentation performed on already compiled classes that do not store information from where they were taken: from a dependency or from user code.

Therefore, the plugin needs additional information about which class is custom. To do this, the plugin must be modified taking into account Gradle project isolation and configuration caches.

The [issue for this feature](#418) has been created, but it is in low priority, because this is not a very frequent problem and is solved by configuring the `excludeInstrumentation` block

**Comment by shanshin** [comment]
>  But does excludeInstrumentation have any affect?

It will not affect the report in any way if the class from the dependency is specified (in your case from `kotlin-aws-s3`)

> So maybe this issue can be closed.

Yes, this task can be closed in favor of #418
